### ISSUE TITLE
Tests fail when countries list is changed

### ISSUE THREAD

**Issue by Kwaidan00** [mirumee/saleor#3849]
### What I'm trying to achieve

I would like to allow shipping to only one country. When I restricts countries used by `django-countries` by defining `COUNTRIES_ONLY` in settings, about 100 (one hundred) tests failed.

### Steps to reproduce the problem
1. Add `COUNTRIES_ONLY` to `settings.py` (e.g. `COUNTRIES_ONLY = ['US']`)
2. Run `py.test`

### What I expected to happen
I can restrict countries where I ship orders and tests pass.
In addition, I would like to write my own tests to check my countries constraint is used.

### Screenshots

**System information**
Operating system: MacOS 10.13, Python 3.7
Browser: n/a

**Comment by NyanKiyoshi** [comment]
Hi, you should override the value you set in your settings into the `tests/settings.py` file.

**Comment by Kwaidan00** [comment]
@NyanKiyoshi I know I can override it in `tests/settings.py`. But if I did that, how I would write my own tests to check my constraint? How to check the only allowed country is, e.g., US?
 
I think that tests settings should be as close to the real settings as it is possible.

**Comment by NyanKiyoshi** [comment]
@Kwaidan00 no. If you want to test given custom settings, you will want to create a new fixture overriding the settings fixture. For an example of editing the settings fixture, take a look at this following code:
https://github.com/mirumee/saleor/blob/3cfe829cf9babee8da03983177df75dd2ada4d69/tests/test_core.py#L165-L168

**Comment by Kwaidan00** [comment]
@NyanKiyoshi Thanks for your help. I know I can do that way.
Anyway, when I change something in `settings.py`, I should not expect that tests fails and I have to revert my changes in `tests/settings.py`. If tests relay on that setting, maybe it should be explicitly in `tests/settings.py`?

On the other hand, I couldn't just change setting in my test. `countries` object from `django-countries` caches calculated available countries. 
I wrote that test, but I had to use `delattr` on `countries` object and I really don't like it...
